### PR TITLE
Agile CRM - new sources & actions

### LINKED ISSUE #3544: Agile CRM
### App
Agile CRM

### Triggers
* New Contact
* New Ticket
* New Task

### Actions
* Create Ticket
* Create Task
* Create or Update Contact

### PR DESCRIPTION
**Pull request by michelle0927**
Resolves #3544 

**Sources**
- New Contact
- New Ticket
- New Task

**Actions**
- Create Ticket
- Create Task
- Create Contact
- Update Contact

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4134"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/agile_crm/agile_crm.app.mjs [hunk 1]
```diff
@@ -1,11 +1,252 @@
...
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _getBaseUrl() {
+      return `https://${this.$auth.domain}.agilecrm.com/dev/api/`;
+    },
+    _getAuth() {
+      return {
+        username: this.$auth.username,
+        password: this.$auth.api_key,
```
Comment (lcaresia) [inline components/agile_crm/agile_crm.app.mjs:173]: Move this to a getter method;

```suggestion
        username: this._username(), // this.$auth.username,
        password:  this.apiKey(), // this.$auth.api_key,
```

File: components/agile_crm/sources/common/base.mjs [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import agileCrm from "../../agile_crm.app.mjs";
+
+export default {
+  props: {
+    agileCrm,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
```
Comment (lcaresia) [inline components/agile_crm/sources/common/base.mjs:10]:
```suggestion
        intervalSeconds: 60 * 15, // 15 minutes
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Thanks for your contribution.

I requested just some fixes!

**Review by lcaresia — DISMISSED** [review]
LGTM!

**Comment by DilanAthukorala** [comment]
Hi @michelle0927,

For "Update Contact", is "First Name" param mandatory? If not please make it an optional parameter
image

**Comment by DilanAthukorala** [comment]
Verification completed and ready for release.

**Review by lcaresia — APPROVED** [review]

**Pull request merged**